quinn-proto/src/connection/datagrams.rs

```diff
@@ -68,11 +68,19 @@ impl<'a> Datagrams<'a> {
     ///
     /// Not necessarily the maximum size of received datagrams.
     pub fn max_size(&self) -> Option<usize> {
+        let key = self.conn.spaces[SpaceId::Data].crypto.as_ref().map_or_else(
```

Comment:
Would be nice to split the large expression in a separate commit and then fix the panic from there.

I suspect that, as in the other PR, `match` would be easier to follow than `map_or_else()`.